### PR TITLE
Added streamline filtering by primary orientation

### PR DESCRIPTION
**Pull request by jyeatman**
Added a function to filter streamlines by their primary orientation. We now need to add that to the bundle dict as a criteria

### CODE REVIEW

File: AFQ/segmentation.py [hunk 1]
```diff
@@ -1183,6 +1183,23 @@ def _is_streamline_in_ROIs_parallel(indiv_args, tol, include_roi,
...
+    """
+    Compute the cardinal orientation of each streamline
+    Parmaeters
+    __________
+    streamlines : sequence of 3XN_i arrays
+    """
+
+    axis_diff = np.zeros((len(streamlines),3))
+    for ii, sl in enumerate(streamlines):
+        axis_diff[ii,:] = np.abs(sl[0,:] - sl[-1,:])
```
Comment (arokem) [inline AFQ/segmentation.py:1197]:
```suggestion
        axis_diff[ii, :] = np.abs(sl[0, :] - sl[-1, :])
```

Comment (arokem) [reply AFQ/segmentation.py:1197]: PEP8

**Review by arokem — COMMENTED** [review]
Very nice! Would be great to add a test.

**Comment by arokem** [comment]
IIUC, this currently decides whether a streamline is vertical based on the start and end points. But you could imagine a streamline that curves around horizontally for most of its length and gets filtered here as vertical if its start and end points are slightly vertical.

**Comment by 36000** [comment]
Superseded by #898

**Pull request closed without merging**